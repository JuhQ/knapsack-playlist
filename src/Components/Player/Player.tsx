import React from "react";
import YouTube from "react-youtube";

interface Props {
  id: string;
  onEnd: () => void;
}

const Player: React.FC<Props> = ({ id, onEnd }: Props) => (
  <YouTube videoId={id} onEnd={onEnd} opts={{ playerVars: { autoplay: 1 } }} />
);

export default Player;
