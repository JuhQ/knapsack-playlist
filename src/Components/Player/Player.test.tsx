import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import Player from "./Player";

describe("Player", () => {
  it("should render", () => {
    const component = renderer.create(<Player id="1" onEnd={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be able to call onEnd function", () => {
    const handleOnEnd = jest.fn();
    const wrapper = shallow(<Player id="1" onEnd={handleOnEnd} />);
    expect(handleOnEnd).not.toHaveBeenCalled();
    wrapper.simulate("end");
    expect(handleOnEnd).toHaveBeenCalled();
  });
});
