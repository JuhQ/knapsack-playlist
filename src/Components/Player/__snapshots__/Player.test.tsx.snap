// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should render 1`] = `
<div
  className=""
>
  <div
    className={null}
    id={null}
  />
</div>
`;
